### ISSUE TITLE
LoweringError when passing a tuple to numpy with under @njit(parallel=True) mode

### ISSUE THREAD

**Issue by 0xzz** [numba/numba#4277]
This is possibly a duplicate with #2625.

In #2625, a LoweringError exception is raised when passing a tuple argument to prange in parallel mode. The same exception is raised in the following minimal example:

`
import numpy as np
from numba import njit

shape = (3, 3, 3)
@njit(parallel=True)
def test_tuple_argument():
    m = np.ones(shape)
    return m.sum()

test_tuple_argument()
`

**Comment by esc** [comment]
@Oxzz thank you for reporting this and helping to improve Numba. I also believe that this is a duplicate and will mark it accordingly.

**Comment by jl-wynen** [comment]
I don't think this really is a duplicate of #2625.
I am using Python 3.8.1 with numba 0.48.0.dev0+224.g40fb2d068 and the code shown in #2625 works fine but the bug reported here still exists.

In case it helps, here is an updated error message
```
numba.errors.LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
'$8call_method.3_size0.28'
File "numba_bug.py", line 21:
def test_tuple_argument():
    m = np.ones(shape)
    ^
[1] During: lowering "id=0[LoopNest(index_variable = parfor_index.37, range = (0, $8call_method.3_size0.28, 1)), LoopNest(index_variable = parfor_index.38, range = (0, $8call_method.3_size1.29, 1)), LoopNest(index_variable = parfor_index.39, range = (0, $8call_method.3_size2.30, 1))]{48: <ir.Block at /home/jl/Uni/PhD/cns/foldilearn/numba_bug.py (21)>}Var($parfor_index_tuple_var.45, numba_bug.py:21)" at /home/jl/Uni/PhD/cns/foldilearn/numba_bug.py (21)
```

The same bug appears when passing `shape` as a function argument to `test_tuple_argument` but not when specifying it in the function itself via `m = np.ones((3, 3, 3))`.

**Comment by stuartarchibald** [comment]
I don't think that the problem reported here is related to #2625. The root issue is from:
```
numba/lowering.py", line 1209, in getvar
    return self.varmap[name]
KeyError: '$8call_method.3_size0.28'
```
whereas #2625 is a known and caught/handled problem to do with tuples as args to gufuncs.

CC @DrTodd13
